plugin/secrets_service_account_key.go

```diff
@@ -159,6 +162,53 @@ func (b *backend) secretKeyRevoke(ctx context.Context, req *logical.Request, d *
 	if !ok {
 		return nil, fmt.Errorf("secret is missing key_name internal data")
 	}
+	keyName := keyNameRaw.(string)
+
+	rolesetNameRaw, ok := req.Secret.InternalData["role_set"]
+	if !ok {
+		return nil, fmt.Errorf("secret is missing role_set internal data")
+	}
+	rolesetName := rolesetNameRaw.(string)
+
+	b.saCacheLock.Lock()
+	defer b.saCacheLock.Unlock()
+
+	cacheCollection, err := getCacheCollection(ctx, req.Storage, rolesetName)
+	if err != nil {
+		return logical.ErrorResponse(fmt.Sprintf("unable to get service account key cache collection of role %s: %v", rolesetName, err)), nil
+	}
+
+	item := cacheCollection.Items[keyName]
+	item.Counter--
+
+	if item.Counter == 0 {
+		delete(cacheCollection.Items, keyName)
```

Comment:
I think `cacheCollection.Delete(keyname)` should be a method of `serviceAccountKeyCacheCollection`

Reply:
Noted. I'll update it in the next commit.